### PR TITLE
Upgrade compiler version to 0.8.19

### PR DESCRIPTION
**Pull request by FlattestWhite**
# Code Review Processes
## New Feature Review
Before submitting a pull request for new review, make sure the following is done:
* Design doc is created and posted here: [Insert Link]
* Code cleanliness and completeness is addressed via [guidelines](https://app.gitbook.com/@setprotocol-1/s/set/smart-contract-engineering/sc-code-review-process)

README Checks
- [] README has proper context for the reviewer to understand what the code includes, any important design considerations, and areas to pay more attention to

Code Checks
- [] Add explanatory comments. If there is complex code that requires specific context or understanding, note that in a comment
- [] Remove unncessary comments. Any comments that do not add additional context, information, etc. should be removed
- [] Add javadocs. 
- [] Scrub through the code for inconsistencies (e.g. removing extra spaces)
- [] Ensure there are not any .onlys in spec files

Broader Considerations
- [] Ensure variable, function and event naming is clear, consistent, and reflective for the scope of the code.
- [] Consider if certain pieces of logic should be placed in a different library, module

### CODE REVIEW

**Comment by pblivin0x** [comment]
things to try
+ binary search
+ compile one --> only keep core contracts in `dev` branch

**Comment by pblivin0x** [comment]
🔥

**Comment by pblivin0x** [comment]
Let's update the target branch for this PR from `master` to a new `dev` branch

File: contracts/lib/PreciseUnitMath.sol [hunk 1]
```diff
@@ -13,15 +13,15 @@
...
-pragma solidity 0.6.10;
-pragma experimental ABIEncoderV2;
+pragma solidity 0.8.19;
 
-import { SafeCast } from "@openzeppelin/contracts/utils/SafeCast.sol";
-import { SafeMath } from "@openzeppelin/contracts/math/SafeMath.sol";
-import { SignedSafeMath } from "@openzeppelin/contracts/math/SignedSafeMath.sol";
+
+import { SafeCast } from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import { SafeMath } from "@openzeppelin/contracts/utils/math/SafeMath.sol";
```
Comment (pblivin0x) [inline contracts/lib/PreciseUnitMath.sol:23]: Do we still need `SafeMath` with compiler 0.8 ? Could create a huge, unnecessary diff though.

Comment (FlattestWhite) [reply contracts/lib/PreciseUnitMath.sol:23]: We could remove SafeMath as the overflow/underflow checks are now inbuilt. The SafeMath lib now provides nice abstractions like tryAdd etc. That can be a future TODO to remove SafeMath where possible

**Comment by FlattestWhite** [comment]
> Let's update the target branch for this PR from `master` to a new `dev` branch

Will you setup the new dev branch?
